### PR TITLE
[Data] Unify "From*" operators

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

This PR unifies the definitions of all `From*` operators. Previously, converting input data to blocks was redundantly done both in the `from_*` apis and in the "InputDataBuffer" physical operators. Now, for all the `from_*` apis, we convert the data to blocks first and pass them to the operator. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ollie-iterators** [comment]
Do you think that you can abstract the "From*" operators more by changing the class "FromItems", "FromNumpy", ... to a single class or just use the strings because the classes only return a string that is the same as the name of the class?

**Comment by raulchen** [comment]
> I think that the classes "FromItems" and "FromNumpy" in the file "python/ray/data/_internal/logical/operators/from_operators.py" are unnecessary because they just return a string that is the name of the class.

I've simplified the definition of these classes. But I think it's still useful to keep them. Because it will give us useful info about the data source for "Dataset.stats()"

File: python/ray/data/tests/test_numpy.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def test_from_numpy(ray_start_regular_shared, from_ref):
     values = np.stack(extract_values("data", ds.take(8)))
     np.testing.assert_array_equal(values, np.concatenate((arr1, arr2)))
     # Check that conversion task is included in stats.
-    assert "FromNumpyRefs" in ds.stats()
+    assert "FromNumpy" in ds.stats()
```
Comment (ericl) [inline python/ray/data/tests/test_numpy.py:47]: Shall we keep the "Refs" part of the name as well for consistency?

**Review by ericl — APPROVED** [review]
Maybe consider preserving the FromNumpy vs FromNumpyRefs distinction, but I could see it either way.

**Comment by raulchen** [comment]
> Maybe consider preserving the FromNumpy vs FromNumpyRefs distinction, but I could see it either way.

We didn't distinguish FromNumpy and FromNumpyRefs before either. Because what `from_numpy` does is simply `put` + `from_numpy_refs`. Thus, I think it's not very useful to use `FromNumpyRefs` for the name, and just changed it to `FromNumpy`.

File: python/ray/data/read_api.py [hunk 2]
```diff
@@ -1590,15 +1573,15 @@ def from_pandas_refs(
...
-    logical_plan = LogicalPlan(FromPandasRefs(dfs))
     context = DataContext.get_current()
     if context.enable_pandas_block:
         get_metadata = cached_remote_fn(get_table_block_metadata)
         metadata = ray.get([get_metadata.remote(df) for df in dfs])
+        logical_plan = LogicalPlan(FromPandas(dfs, metadata))
         return MaterializedDataset(
             ExecutionPlan(
                 BlockList(dfs, metadata, owned_by_consumer=False),
-                DatasetStats(stages={"FromPandasRefs": metadata}, parent=None),
```
Comment (scottjlee) [inline python/ray/data/read_api.py:1601]: since this is still under `from_pandas_refs`, should the stats be using the same label here?

**Review by scottjlee — APPROVED** [review]

File: python/ray/data/read_api.py (hunk 2 above)
Comment (scottjlee) [reply python/ray/data/read_api.py:1601]: ah i just saw your reply to Eric's comment above. that makes sense to me. feel free to leave out the `refs` part of the label

**Pull request merged**